rust/trace-dump/src/chrome_trace.rs

```diff
@@ -29,58 +26,175 @@ pub enum OutputFormat {
 #[derive(Debug)]
 pub enum Error {
     Io(IOError),
-    Json(serde_json::Error)
+    Json(serde_json::Error),
+    DecodingFormat(String)
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+struct ChromeTraceArrayEntryArgs {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    xi_sample_id: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    payload: Option<TracePayloadT>,
+}
+
+/// This is the struct that represents a single entry within an array of
+/// samples.  Vec<ChromeTraceArrayEntry> would represent a single complete
+/// trace.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+struct ChromeTraceArrayEntry {
+    name: StrCow,
+    cat: StrCow,
+    ph: StrCow,
+    #[serde(rename = "ts")]
+    ts_us: u64,
+    pid: u64,
+    tid: u64,
+    args: Option<ChromeTraceArrayEntryArgs>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+enum ChromeTraceArrayEntries {
```

Comment:
Is there a reason this is an enum, and not just a tuple struct?

Reply:
Well Chrome trace does technically support 2 serialization formats.  I think I'm overthinking things here too but all this should disappear in the cleanup commit where i'll just change the Sample to map 1:1 with chrome trace events.